runner/internal/reflectutil/reflectutil.go

```diff
@@ -0,0 +1,171 @@
+package reflectutil
+
+import (
+	"fmt"
+	"reflect"
+	"strconv"
+	"strings"
+)
+
+// KeyMatcher is a function that determines whether a property name
+// matches the current path value.
+type KeyMatcher func(key, pathname string) bool
+
+// PathResolver exposes a method ResolvePath that resolves a path
+// representation from a host structure.
+type PathResolver struct {
+	// KeyMatcher determines whether a structure key matches the current
+	// path value. If not set, regular string comparison is used.
+	KeyMatcher KeyMatcher
+}
+
+// ResolvePath resolves pathRepr starting from host and returns
+// the corresponding value.
+func (r PathResolver) ResolvePath(host interface{}, pathRepr string) reflect.Value {
```

Comment:
Suggestion to mirror the new method:

```suggestion
// ResolvePathValue resolves pathRepr starting from host and returns
// the corresponding value.
func (r PathResolver) ResolvePathValue(host interface{}, pathRepr string) reflect.Value {
```